parquet-hadoop/src/test/java/org/apache/parquet/format/converter/TestParquetMetadataConverter.java

```diff
@@ -252,4 +267,47 @@ public void randomTestFilterMetaData() {
     }
   }
 
+  @Test
+  public void testNullFieldMetadataDebugLogging() throws NoSuchFieldException, IllegalAccessException, IOException {
+    Path testFile = new Path("target/test/metadata_debug_log_bug.parquet");
+    Configuration conf = new Configuration();
+
+    deleteFile(testFile, conf);
+
+    createTestFile(testFile, conf);
+
+    //enable debug logging
+    Field debugFlag = Log.class.getField("DEBUG");
+    Field modifiersField = Field.class.getDeclaredField("modifiers");
+    modifiersField.setAccessible(true);
+    modifiersField.setInt(debugFlag, debugFlag.getModifiers() & ~Modifier.FINAL);
```

Comment:
Yep I will simplify this. This attempt was just to create the exact scenario of how a user will hit this problem.